### ISSUE TITLE
Bad error message on `.await` + `?`

### ISSUE THREAD

**Issue by taiki-e** [taiki-e/futures-async-stream#6]
Code:
```rust
#![feature(async_await, stmt_expr_attributes, proc_macro_hygiene)]
use futures::stream;
use futures_async_stream::for_await;

async fn foo() {
    #[for_await]
    for _i in stream::iter(1..2) {
        async {}.await?;
    }
}
```

Error message:
```text
error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
 --> tests\async_stream.rs:5:5
  |
5 |     #[for_await]
  |     ^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
  |
  = help: the trait `std::ops::Try` is not implemented for `()`
  = note: required by `std::ops::Try::into_result`
```

This error message can be very confusing to the user if more than one `?` operator is used.

**Comment by taiki-e** [comment]
Another case (`.await?` in `#[async_stream]`):

Code:
```rust
#[async_stream(item = i32)]
async fn async_stream_fn() {
    for _i in 1..2 {
        async {}.await?;
    }
}
```
Error message:

```text
error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> tests\async_stream.rs:13:1
   |
13 | #[async_stream(item = i32)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`
```

This fixed in #7.

**Comment by taiki-e** [comment]
Unfortunately, this is a compiler bug and probably the fix will require https://github.com/rust-lang/rust/issues/43081.
